crates/air-parser/src/parser/lexer/errors.rs

```diff
@@ -32,6 +35,20 @@ pub enum LexerError {
 
     #[error("invalid character in json path")]
     InvalidJsonPath(usize, usize),
+
+    #[error("this signed digit could have only digits or one dot")]
+    UnallowedCharInNumber(usize, usize),
+
+    #[error("{2}")]
+    ParseIntError(usize, usize, ParseIntError),
+
+    #[error("{2}")]
+    ParseFloatError(usize, usize, ParseFloatError),
```

Comment:
use `#[source]` for error chaining:

```suggestion
    ParseFloatError(usize, usize,  #[source] ParseFloatError),
```

Reply:
Are there any benefits of using `#[source]` inside a enum?